staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -407,6 +412,81 @@ func (obj InfoObject) Name() string {
 	)
 }
 
+func (obj InfoObject) GroupVersionKind() schema.GroupVersionKind {
+	return obj.Info.Mapping.GroupVersionKind
+}
+
+// mask conceals any secret values and returns the masked from/to versions
+// of the object.
+//
+// All secret values in the objects will be masked with a fixed-length
+// asterisk mask. If two values are different, an additional suffix will
+// be added so they can be diffed.
+func mask(from, to runtime.Object) (runtime.Object, runtime.Object, error) {
+	// Extract nested map object
+	unstructLive, dataLive, err := unstructuredNestedMap(from, "data")
```

Comment:
Could you decode these directly into a Secret object since we know exactly what they are? I think it'd keep the code a little simpler.

Reply:
The arbitrary "data" isn't useful since we know the path. Also, there are too many return variables, please remove 1. :-)

Reply:
@apelisse So I've tried to convert it to the `corev1.Secret` object but the masked values that I've patched in was encoded as base64 on output when running `kubectl diff`, which is somewhat unsurprising to me. Any suggestions on this?

Reply:
Yeah, the data is base64'd, so you can't really print whatever you want :-(. Let me see.